wire-ios/Wire-iOS Tests/SwiftSnapshotTesting/SnapshotHelper.swift

```diff
@@ -138,7 +138,54 @@ struct SnapshotHelper {
         XCTAssertNil(failure, file: file, line: line)
     }
 
-    private func snapshotDirectory(file: StaticString = #file) -> String {
+    /// Verifies that a given `UIView` renders correctly across all supported Dynamic Type content size categories.
+    ///
+    /// - Parameters:
+    ///   - value: The `UIView` instance that you want to verify.
+    ///   - name: The name of the reference image.
+    ///   - file: The invoking file name.
+    ///   - testName: The name of the reference image.
+    ///   - line: The invoking line number.
+
+    func verifyForDynamicType(
+        matching value: UIView,
+        named name: String? = nil,
+        file: StaticString = #file,
+        testName: String = #function,
+        line: UInt = #line
+    ) {
+        [
+            "extra-small": UIContentSizeCategory.extraSmall,
+            "small": .small,
+            "medium": .medium,
+            "large": .large,
+            "extra-large": .extraLarge,
+            "extra-extra-large": .extraExtraLarge,
+            "extra-extra-extra-large": .extraExtraExtraLarge,
+            "accessibility-medium": .accessibilityMedium,
+            "accessibility-large": .accessibilityLarge,
+            "accessibility-extra-large": .accessibilityExtraLarge,
+            "accessibility-extra-extra-large": .accessibilityExtraExtraLarge,
+            "accessibility-extra-extra-extra-large": .accessibilityExtraExtraExtraLarge
+        ].forEach { name, contentSize in
+            let failure = verifySnapshot(
+                matching: value,
+                as: .image(
+                    traits: .init(preferredContentSizeCategory: contentSize)
+                ),
+                named: name,
+                snapshotDirectory: snapshotDirectory(file: file),
+                file: file,
+                testName: testName,
+                line: line
+            )
```

Comment:
It would be nice to use the existing verify method if possible.

Reply:
What do you mean by existing?

Reply:
```swift
let failure = verifySnapshot(
            matching: value,
            as: .image(
                perceptualPrecision: perceptualPrecision,
                layout: layout,
                traits: traits
            ),
            snapshotDirectory: snapshotDirectory(file: file),
            file: file,
            testName: testName,
            line: line
        )
```

if I try and do something like that I get an error 

`Referencing static method 'image(drawHierarchyInKeyWindow:precision:perceptualPrecision:layout:traits:)' on 'Snapshotting' requires that 'UIViewController' conform to 'View'`

I have a feeling that, this image init is only available when we would like to test SwiftUI views

Reply:
nvm I made it work. the problem was including layout. layout only works for methods related with SwiftUI. 

https://github.com/wireapp/wire-ios/commit/b569bd280263c125105f73db09591b3d71bdf590